rero_ils/modules/libraries/api.py

```diff
@@ -117,39 +118,33 @@ def get_email(self, notification_type):
                 if setting['type'] == notification_type:
                     return setting['email']
 
-    def _pickup_location_query(self):
-        """Search the location index for pickup locations."""
-        return LocationsSearch() \
-            .filter('term', library__pid=self.pid) \
-            .filter('term', is_pickup=True) \
-            .source(['pid']) \
-            .scan()
-
     def location_pids(self):
-        """Return a generator of ES Hits of all pids of library locations."""
-        return LocationsSearch() \
+        """Return location pids related to this library."""
+        query = LocationsSearch() \
             .filter('term', library__pid=self.pid) \
-            .source(['pid']) \
-            .scan()
+            .source('pid')
+        return [hit.pid for hit in query.scan()]
 
-    def get_pickup_locations_pids(self):
+    @cached_property
```

Comment:
What's happens when a location pickup property is modified?

Reply:
Using `@cached_property` the return value of the function is cached until end-of-life for current object (or cached is specifically cleared). So it could cause problem if the pickup location changed during `Library` life cycle. But in practice this change only with UI and REST-API and it's seems safe for RERO-ILS usage.